src/wasm-interpreter.h

```diff
@@ -149,52 +149,13 @@ class Indenter {
 // Execute an expression
 template<typename SubType>
 class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
-public:
-  enum FlagValues {
-    // By default, just evaluate the expression, i.e. all we want to know is
-    // whether it computes down to a concrete value, where it is not necessary
-    // to preserve side effects like those of a `local.tee`.
-    DEFAULT = 0,
-    // Be very careful to preserve any side effects. For example, if we are
-    // intending to replace the expression with a constant afterwards, even if
-    // we can technically evaluate down to a constant, we still cannot replace
-    // the expression if it also sets a local, which must be preserved in this
-    // scenario so subsequent code keeps functioning.
-    PRESERVE_SIDEEFFECTS = 1 << 0,
-    // Traverse through function calls, attempting to compute their concrete
-    // value. Must not be used in function-parallel scenarios, where the called
-    // function might be concurrently modified, leading to undefined behavior.
-    TRAVERSE_CALLS = 1 << 1
-  };
-
-  // Flags indicating special requirements, for example whether we are just
-  // evaluating (default), also going to replace the expression afterwards or
-  // executing in a function-parallel scenario. See FlagValues.
-  typedef uint32_t Flags;
-
-  // Indicates no limit of maxDepth or maxLoopIterations.
-  static const Index NO_LIMIT = 0;
-
 protected:
-  // Optional module context to search for globals and called functions. NULL if
-  // we are not interested in any context or if the subclass uses an alternative
-  // mechanism, like RuntimeExpressionRunner does.
-  Module* module = nullptr;
-
-  // Flags indicating special requirements. See FlagValues.
-  Flags flags = FlagValues::DEFAULT;
-
   // Maximum depth before giving up.
-  Index maxDepth = NO_LIMIT;
+  Index maxDepth;
```

Comment:
In https://github.com/WebAssembly/binaryen/pull/2702 it has been requested to initialize both ´maxDepth` and `maxLoopIterations` here with `= NO_LIMIT``. However, the single constructor now sets these while allowing to omit either just `maxLoopIterations` or both. So, C++ newbie question: Are there scenarios where a derived template may never call that constructor?